WooCommerce/UITestsFoundation/Screens/Products/ProductSearchScreen.swift

```diff
@@ -0,0 +1,47 @@
+import ScreenObject
+import XCTest
+
+public final class ProductSearchScreen: ScreenObject {
+
+    private let searchTextFieldGetter: (XCUIApplication) -> XCUIElement = {
+        $0.descendants(matching: .other).element(matching: .any, identifier: "product-search-screen-search-field")
+    }
+
+    private var searchTextField: XCUIElement { searchTextFieldGetter(app) }
+    private var updatedSearchResultsID = "updated-search-results-table-view"
+    public static let defaultSearchResultsID = "default-search-results-table-view"
+
+    public init(app: XCUIApplication = XCUIApplication()) throws {
+        try super.init(
+            expectedElementGetters: [ searchTextFieldGetter ],
+            app: app
+        )
+    }
+
+    public func enterSearchCriteria(text: String) throws -> Self {
+        searchTextField.tap()
+        searchTextField.enterText(text: text)
+        return self
+    }
+
+    @discardableResult
+    public func verifyNumberOfProductsOnSearchResults(expectedNumberOfProducts: Int, accessibilityId: String? = defaultSearchResultsID) throws -> Self {
+        let productTableView = app.tables.matching(identifier: accessibilityId!)
```

Comment:
💡  Since both `updatedSearchResultsID` and `defaultSearchResultsID` end with `-search-results-table-view`, this will work:

```suggestion
        let productTableView = app.tables.matching(NSPredicate(format: "identifier CONTAINS '-search-results-table-view'"))
```

Pros:
- No need to have the 2nd argument for the function and care about which identifier to use
- `updatedSearchResultsID` and `defaultSearchResultsID` can be removed

Cons:
- Loosing a bit of robustness.

Reply:
Thanks for the suggestion! But for this specific case, I will leave this as is; initially, I used a different accessibility ID but found that it appears on both views (the old and new views are displayed), making the test incorrect. 

I'd like to keep the unique ID instead of a generic one so that we can be sure that the results are from the correct table view.